src/test/java/org/opensearch/knn/plugin/script/PainlessScriptIT.java

```diff
@@ -126,22 +126,42 @@ private List<MappingProperty> buildMappingProperties() {
 
     public void testL2ScriptScoreFails() throws Exception {
         String source = String.format("1/(1 + l2Squared([1.0f, 1.0f], doc['%s']))", FIELD_NAME);
-        Request request = buildPainlessScriptRequest(source, 3, getL2TestData());
+        Request request = buildPainlessScoreScriptRequest(source, 3, getL2TestData());
         addDocWithNumericField(INDEX_NAME, "100", NUMERIC_INDEX_FIELD_NAME, 1000);
         expectThrows(ResponseException.class, () -> client().performRequest(request));
         deleteKNNIndex(INDEX_NAME);
     }
 
-    private Request buildPainlessScriptRequest(String source, int size, Map<String, Float[]> documents) throws Exception {
+    private Request buildPainlessScoreScriptRequest(String source, int size, Map<String, Float[]> documents) throws Exception {
         buildTestIndex(documents);
         QueryBuilder qb = new MatchAllQueryBuilder();
-        return constructScriptQueryRequest(INDEX_NAME, qb, Collections.emptyMap(), Script.DEFAULT_SCRIPT_LANG, source, size);
+        return constructScoreScriptQueryRequest(INDEX_NAME, qb, Collections.emptyMap(), Script.DEFAULT_SCRIPT_LANG, source, size);
+    }
+
+    private Request buildPainlessScriptedMetricRequest(
+        String initScriptSource,
+        String mapScriptSource,
+        String combineScriptSource,
+        String reduceScriptSource,
+        Map<String, Float[]> documents,
+        String aggName
+    ) throws Exception {
+        buildTestIndex(documents);
+        return constructScriptedMetricAggregationRequest(
+            aggName,
+            Script.DEFAULT_SCRIPT_LANG,
+            initScriptSource,
+            mapScriptSource,
+            combineScriptSource,
+            reduceScriptSource,
+            documents.size()
+        );
     }
 
     public void testL2ScriptScore() throws Exception {
 
         String source = String.format("1/(1 + l2Squared([1.0f, 1.0f], doc['%s']))", FIELD_NAME);
-        Request request = buildPainlessScriptRequest(source, 3, getL2TestData());
+        Request request = buildPainlessScoreScriptRequest(source, 3, getL2TestData());
```

Comment:
can we change magic number to a constant here?

Reply:
This method was actually creating PainlessScriptRequest with ScoreContext. Since i am adding new request, i renamed accordingly.

Reply:
Here i only renamed the method name. We can do that in generic PR since we need to make this change in many other places.

Reply:
Do you mean separate PR for refactoring/code cleanup? Yes, that makes sense

Reply:
Yes